File: squares-sorted-array.php
<?php

class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function sortedSquares($nums) {
        $new_arr = [];
        $left = 0;
        $right = count($nums)-1;
        $index = count($nums);
        while( $index > 0 ) {
            if( abs($nums[$left]) > abs($nums[$right]) ) {
                $new_arr[(--$index)] = $nums[$left] * $nums[$left];
                $left++;
            } else {
                $new_arr[(--$index)] = $nums[$right] * $nums[$right];
                $right--;
            }
        }

        $arr = [];
        foreach( range( 0, count($nums)-1) as $k )
            $arr[$k] = $new_arr[$k];

        return $arr;
    }
}

$nums  = [-4,-1,0,3,10];
$sol = new Solution();
$ret = $sol->sortedSquares($nums);
echo "<pre>";
print_r( $ret );

File: squares-sorted-array.py
class Solution:
    def sortedSquares(self, nums: list[int]) -> list[int]:
        ret = [0 for x in nums]
        res = [0 for x in nums]
        left = 0
        right = len(nums) - 1
        index = len(nums) - 1

        while index >= 0:
            print(index)
            if abs( nums[left]) > abs(nums[right]):
                ret[index] = nums[left] * nums[left]
                left += 1
            else:
                ret[index] = nums[right] * nums[right]
                right -= 1
            index -= 1

        for i in range(len(nums)):
            res[i] = ret[i]

        return res

nums = [-7,-3,2,3,11] #[-4,-1,0,3,10]
sol = Solution()
ret = sol.sortedSquares(nums)
print(ret)